supertokens_python/recipe/webauthn/interfaces/recipe.py

```diff
@@ -0,0 +1,616 @@
+        return rui.get_as_string()
+
+
+SignInErrorResponse = Union[
+    VerifyCredentialsErrorResponse,
+    LinkingToSessionUserFailedError,
+]
+
+
+class GenerateRecoverAccountTokenResponse(OkResponseBaseModel):
+    token: str
+
+
+GenerateRecoverAccountTokenErrorResponse = UnknownUserIdErrorResponse
+
+
+class ConsumeRecoverAccountTokenResponse(OkResponseBaseModel):
+    email: str
+    user_id: str
+
+
+ConsumeRecoverAccountTokenErrorResponse = RecoverAccountTokenInvalidErrorResponse
+
+
+RegisterCredentialErrorResponse = Union[
+    InvalidCredentialsErrorResponse,
+    OptionsNotFoundErrorResponse,
+    InvalidOptionsErrorResponse,
+    InvalidAuthenticatorErrorResponse,
+]
+
+
+class GetUserFromRecoverAccountTokenResponse(OkResponseBaseModel):
+    user: User
+    recipe_user_id: Optional[RecipeUserId]
+
+    @field_serializer("user")
+    def serialize_user(self, user: User):
+        return user.to_json()
+
+    @field_serializer("recipe_user_id")
+    def serialize_recipe_user_id(self, rui: Optional[RecipeUserId]):
+        if rui is None:
+            return None
+
+        return rui.get_as_string()
+
+
+GetUserFromRecoverAccountTokenErrorResponse = RecoverAccountTokenInvalidErrorResponse
+
+RemoveCredentialErrorResponse = CredentialNotFoundErrorResponse
+
+
+class GetCredentialResponse(OkResponseBaseModel):
+    webauthn_credential_id: str
+    relying_party_id: str
+    recipe_user_id: RecipeUserId
+    created_at: int
+
+    @field_serializer("recipe_user_id")
+    def serialize_recipe_user_id(self, rui: RecipeUserId):
+        return rui.get_as_string()
+
+
+GetCredentialErrorResponse = CredentialNotFoundErrorResponse
+
+
+class ListCredentialsResponse(OkResponseBaseModel):
+    class Credential(CamelCaseBaseModel):
+        webauthn_credential_id: str
+        relying_party_id: str
+        recipe_user_id: str
+        created_at: int
+
+    credentials: List[Credential]
+
+
+RemoveGeneratedOptionsErrorResponse = OptionsNotFoundErrorResponse
+
+
+class GetGeneratedOptionsResponse(OkResponseBaseModel):
+    webauthn_generated_options_id: str
+    relying_party_id: str
+    relying_party_name: str
+    user_verification: UserVerification
+    user_presence: bool
+    origin: str
+    email: Optional[str]
+    timeout: int
+    challenge: str
+    created_at: int
+    expires_at: int
+
+
+GetGeneratedOptionsErrorResponse = OptionsNotFoundErrorResponse
+
+
+UpdateUserEmailErrorResponse = Union[
+    EmailAlreadyExistsErrorResponse,
+    UnknownUserIdErrorResponse,
+]
+
+
+class CreateRecoverAccountLinkResponse(OkResponseBaseModel):
+    link: str
+
+
+class RecoverAccountTokenInput(TypedDict):
+    recover_account_token: str
+
+
+class DisplayNameEmailInput(TypedDict):
+    display_name: Optional[str]
+    email: str
+
+
+class RegisterOptionsKwargsInput(TypedDict):
+    recover_account_token: NotRequired[str]
+    display_name: NotRequired[str]
+    email: NotRequired[str]
```

Comment:
Do u need to validate that only recover_account_token is passed or the other 2, but not all three at the same time?

Reply:
I saw the validation